### PR TITLE
warnings + clippy

### PR DESCRIPTION
**Pull request by TDecki**
- [ ] I have responded to reviews and made changes where appropriate.
- [x] I have tested the code with `cargo test --all` / `./rust/run_all_checks`.
- [x] I have updated comments / documentation related to the changes I made.
- [x] I have rebased my PR branch onto xi-editor/master.

### CODE REVIEW

File: rust/rope/src/delta.rs [hunk 1]
```diff
@@ -673,7 +673,7 @@ impl<'a, N: NodeInfo> Iterator for DeletionsIter<'a, N> {
 
     fn next(&mut self) -> Option<Self::Item> {
         let mut result = None;
-        while let Some(elem) = self.els_iter.next() {
+        for elem in &mut self.els_iter {
```
Comment (Cogitri) [inline rust/rope/src/delta.rs:676]: Does this really have the same semantics? AFAIC the for loop will loop through all elements that are Some while the while loop will abort on the first element that isn’t Some

Comment (TDecki) [reply rust/rope/src/delta.rs:676]: Yes. The [official documentation](https://doc.rust-lang.org/nightly/std/iter/#for-loops-and-intoiterator) which provides pseudocode for a desugared for-loop explicitly terminates the loop once it encounteres a None.

This is also done on behalf of clippy, which is also a good indicator that these snippets are semantically equivalent.

**Review by cmyr — APPROVED** [review]
yep this all looks reasonable!

**Comment by cmyr** [comment]
Thanks!

**Pull request merged**